sql/core/src/main/scala/org/apache/spark/sql/execution/WindowExec.scala

```diff
@@ -177,6 +177,8 @@ case class WindowExec(
         case e @ WindowExpression(function, spec) =>
           val frame = spec.frameSpecification.asInstanceOf[SpecifiedWindowFrame]
           function match {
+            case MakeDecimal(AggregateExpression(f, _, _, _), prec, scale) =>
```

Comment:
When does a `WindowExpression` contain a `MakeDecimal`? As I understand your code correctly, the following things can happen:
##### Sum
The following input:
```Scala
WindowExpression(AggregateExpression(Sum(x), ...), spec)
```
is rewritten into:
```scala
MakeDecimal(WindowExpression(AggregateExpression(Sum(UnscaledValue(x)), ...), spec), prec + 10, scale)
```
##### Average
The following input:
```Scala
WindowExpression(AggregateExpression(Average(x), ...), spec)
```
is rewritten into:
```scala
Cast(Divide(WindowExpression(AggregateExpression(Average(UnscaledValue(x)), ...), spec), Literal.create(math.pow(10.0, scale), DoubleType)), DecimalType(prec + 4, scale + 4))
```